ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/PullQueryConsistencyFunctionalTest.java

```diff
@@ -294,57 +305,242 @@ public void cleanUp() {
+        HighAvailabilityTestUtil.lagsReported(3), USER_CREDS);
     waitForRemoteServerToChangeStatus(
-      clusterFormation.router.getApp(),
-      clusterFormation.standBy.getHost(),
-      HighAvailabilityTestUtil::remoteServerIsUp, USER_CREDS);
+        clusterFormation.router.getApp(),
+        clusterFormation.active.getHost(),
+        HighAvailabilityTestUtil::remoteServerIsUp, USER_CREDS);
+    waitForRemoteServerToChangeStatus(
+        clusterFormation.router.getApp(),
+        clusterFormation.standBy.getHost(),
+        HighAvailabilityTestUtil::remoteServerIsUp, USER_CREDS);
+    final String serializedCV = CONSISTENCY_OFFSET_VECTOR_AFTER.serialize();
+    final ImmutableMap.Builder<String, Object> builder = new ImmutableMap.Builder<String, Object>()
+        .put(KSQL_QUERY_PULL_CONSISTENCY_OFFSET_VECTOR_ENABLED, true)
+        .put(KsqlRequestConfig.KSQL_REQUEST_QUERY_PULL_CONSISTENCY_OFFSET_VECTOR, serializedCV);
+    final Map<String, Object> requestProperties = builder.build();
 
     // When:
+    final Supplier<List<String>> call = () -> {
+      final String response = rawRestQueryRequest(
+          clusterFormation.router.getApp(),
+          sql,
+          MediaType.APPLICATION_JSON,
+          Collections.emptyMap(),
+          requestProperties
+      );
+      return Arrays.asList(response.split(System.lineSeparator()));
+    };
+
+    // Then:
+    final List<String> messages = assertThatEventually(call, hasSize(HEADER + 1));
+    assertThat(messages, hasSize(HEADER + 1));
+    assertThat(messages.get(1), containsString("Failed to get value from materialized table, "
+                                                   + "reason: NOT_UP_TO_BOUND"));
+  }
+
+  @Test
+  public void shouldExecuteThenFailPullQueryWithBound() throws Exception {
+    // Given:
+    ClusterFormation clusterFormation = findClusterFormation(TEST_APP_0, TEST_APP_1, TEST_APP_2);
+    waitForClusterToBeDiscovered(clusterFormation.router.getApp(), 3, USER_CREDS);
+    waitForRemoteServerToChangeStatus(
+        clusterFormation.router.getApp(),
+        clusterFormation.router.getHost(),
+        HighAvailabilityTestUtil.lagsReported(3),
+        USER_CREDS);
+    waitForRemoteServerToChangeStatus(
+        clusterFormation.router.getApp(),
+        clusterFormation.active.getHost(),
+        HighAvailabilityTestUtil::remoteServerIsUp, USER_CREDS);
+    waitForRemoteServerToChangeStatus(
+        clusterFormation.router.getApp(),
+        clusterFormation.standBy.getHost(),
+        HighAvailabilityTestUtil::remoteServerIsUp, USER_CREDS);
+
+    waitForRemoteServerToChangeStatus(
+        clusterFormation.router.getApp(),
+        clusterFormation.router.getHost(),
+        HighAvailabilityTestUtil.lagsReported(clusterFormation.standBy.getHost(), Optional.of(5L), 5),
+        USER_CREDS);
+
+    // Cut off standby from Kafka to simulate lag
+    clusterFormation.standBy.getShutoffs().setKafkaPauseOffset(0);
+    Thread.sleep(2000);
```

Comment:
This is the technique we follow in PullQueryRouting test. On top of my head, I am guessing this is needed to ensure the cutoff actually happened since the cutoff essentially just pauses the consumer